rfd/000?-database-access.md

```diff
@@ -0,0 +1,512 @@
+
+## RBAC
+
+Teleport role resource gets 3 new fields that allow to control access to 
+database instances as well as individual databases and database users.
+
+```yaml
+kind: role
+version: v3
+metadata:
+  name: developer
+spec:
+  allow:
+    # Label selectors for database instances this role has access to.
+    db_labels:
+      environment: ["dev", "stage"]
+    # Database names (within a database instance) this role has access to.
+    db_names: ["main", "metrics", "postgres"]
+    # Database users this role can log in as.
+    db_users: ["alice", "bob"]
+```
+
+## CLI
+
+To connect to a database instance, a user first must login using regular
+`tsh login` command:
+
+```sh
+$ tsh login --proxy=localhost:3080 --user=admin
+```
+
+Once logged in, they can see all available database using the following
+command:
+
+```sh
+$ tsh db ls
+Name            Description     Endpoint                                Labels
+--------------- --------------- --------------------------------------- ---------
+postgres        PostgreSQL 13.0 localhost:5432                          env=dev
+postgres-rds    PostgreSQL 12.4 postgres-rds.xxx.rds.amazonaws.com:5432 env=stage
+postgres-aurora PostgreSQL 11.6 postgres-rds.yyy.rds.amazonaws.com:5432 env=prod
+```
+
+The list will only show databases the logged in user has access to (as per 
+the database labels). For example, developer role shown above won't be able 
+to see the Aurora database instance:
+
+```sh
+$ tsh login --proxy=localhost:3080 --user=dev
+$ tsh db ls
+Name            Description     Endpoint                                Labels
+--------------- --------------- --------------------------------------- ---------
+postgres        PostgreSQL 13.0 localhost:5432                          env=dev
+postgres-rds    PostgreSQL 12.4 postgres-rds.xxx.rds.amazonaws.com:5432 env=stage
+```
+
+To log into a specific database instance, a user executes `tsh db login`
+command. This will retrieve the certificate with encoded database information
+in it:
+
+```sh
+$ tsh db login postgres
+```
+
+The `tsh status` command shows the name of the database a user is logged into:
+
+```sh
+$ tsh status
+> Profile URL:        https://127.0.0.1:3080
+  Logged in as:       dev
+  Roles:              db-developer*
+  ...
+  Database:           postgres
+  ...
+```
+
+The `tsh db login` command also prints a footer explaining how to connect
+to the database:
+
+```sh
+Connection information for "postgres" has been saved to ~/.pg_service.conf.
+You can connect to the database using the following command:
+
+  $ psql "service=postgres user=<user> dbname=<dbname>"
+
+Or configure environment variables and use regular CLI flags:
+
+  $ eval $(tsh db env)
+  $ psql -U <user> <database>
+```
+
+Upon successful login to a PostgreSQL instance, `tsh` configures a [connection
+service file](https://www.postgresql.org/docs/9.1/libpq-pgservice.html) which
+resides in `~/.pg_service.conf`.
```

Comment:
are you going to override users file? they will not be happy if we rewrite their file :)

Reply:
Yeah, of course it won't fully overwrite the file, it will add an ini section to it. If the file already happens to have the section with the same name, it will get overwritten though.

Reply:
add this to the doc